src/cpp/_pythia8.cpp

```diff
@@ -110,6 +110,16 @@ PYBIND11_MODULE(_pythia8, m)
         .def("addParticle", py::overload_cast<int, string, string, int, int, int, double, double, double, double, double, bool>(&ParticleData::addParticle), "pdgid"_a, "name"_a, "antiname"_a, "spinType"_a = 0, "chargeType"_a = 0, "colType"_a = 0, "m0"_a = 0, "mWidth"_a = 0, "mMin"_a = 0, "mMax"_a = 0, "tau0"_a = 0, "varWidth"_a = false)
         .def("isParticle", &ParticleData::isParticle)
         .def("findParticle", py::overload_cast<int>(&ParticleData::findParticle))
+        // TODO better style is .def("__iter__", ...) and return make_value_iterator
+        // here, but this leads to segfault which I cannot fix right now
```

Comment:
I don't know whether it is related here, but I returned iterator for Pythia like this (and it worked):

```
py::class_<Event>(m, "Event")
...
.def("__iter__", [](const Event &event) { return py::make_iterator(event.begin(), event.end()); },
                         py::keep_alive<0, 1>() /* Essential: keep object alive while iterator exists */)
```

Reply:
I tried something like this, including the keep_alive, which I am aware of, but it was segfaulting. I did not have time to figure out why.

Reply:
I checked again, but same result. It is probably because the list contains smart pointers instead of values and somewhere the life-time management goes wrong. pybind11 tries its best to use handle tricky cases like this, but sometimes this has to be handled manually. I don't know how to fix this properly, and since we use this API only internally, I will go over this issue to move forward.